src/effects/backends/builtin/glitcheffect.cpp

```diff
@@ -0,0 +1,126 @@
+    delay->setDescription(QObject::tr(
+            "Delay time\n"
+            "1/8 - 2 beats if tempo is detected\n"
+            "1/8 - 2 seconds if no tempo is detected"));
+    delay->setValueScaler(EffectManifestParameter::ValueScaler::Linear);
+    delay->setUnitsHint(EffectManifestParameter::UnitsHint::Beats);
+    delay->setRange(0.0, 0.5, 2.0);
+
+    EffectManifestParameterPointer quantize = pManifest->addParameter();
+    quantize->setId("quantize");
+    quantize->setName(QObject::tr("Quantize"));
+    quantize->setShortName(QObject::tr("Quantize"));
+    quantize->setDescription(QObject::tr(
+            "Round the Time parameter to the nearest 1/8 beat."));
+    quantize->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    quantize->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    quantize->setRange(0, 1, 1);
+
+    EffectManifestParameterPointer triplet = pManifest->addParameter();
+    triplet->setId("triplet");
+    triplet->setName(QObject::tr("Triplets"));
+    triplet->setShortName(QObject::tr("Triplets"));
+    triplet->setDescription(
+            QObject::tr("When the Quantize parameter is enabled, divide "
+                        "rounded 1/8 beats of Time parameter by 3."));
+    triplet->setValueScaler(EffectManifestParameter::ValueScaler::Toggle);
+    triplet->setUnitsHint(EffectManifestParameter::UnitsHint::Unknown);
+    triplet->setRange(0, 0, 1);
+
+    return pManifest;
+}
+
+void GlitchEffect::loadEngineEffectParameters(
+        const QMap<QString, EngineEffectParameterPointer>& parameters) {
+    m_pDelayParameter = parameters.value("delay_time");
+    m_pQuantizeParameter = parameters.value("quantize");
+    m_pTripletParameter = parameters.value("triplet");
+}
+
+void GlitchEffect::processChannel(
+        GlitchGroupState* pGroupState,
+        const CSAMPLE* pInput,
+        CSAMPLE* pOutput,
+        const mixxx::EngineParameters& engineParameters,
+        const EffectEnableState enableState,
+        const GroupFeatureState& groupFeatures) {
+
+    // The minimum of the parameter is zero so the exact center of the knob is 1 beat.
+    double period = m_pDelayParameter->value();
+
+    int delay_frames;
+    double min_delay;
+    if (groupFeatures.has_beat_length_sec) {
+        if (m_pQuantizeParameter->toBool()) {
+            period = roundToFraction(period, 8);
+            if (m_pTripletParameter->toBool()) {
+                period /= 3.0;
+            }
+        }
+        period = std::max(period, 1 / 8.0);
+        delay_frames = static_cast<int>(period * groupFeatures.beat_length_sec *
+                engineParameters.sampleRate());
+        min_delay = 1 / 8.0 * groupFeatures.beat_length_sec * engineParameters.sampleRate();
+    } else {
+        delay_frames = static_cast<int>(period * engineParameters.sampleRate());
+        min_delay = 1 / 8.0 * engineParameters.sampleRate();
+    }
+
+    if (delay_frames < min_delay) {
+        delay_frames = 0;
```

Comment:
This can just be replaced by the content of the other if:
```suggestion
        SampleUtil::copy(
                pOutput,
                pInput,
                engineParameters.samplesPerBuffer());
        return;
```

Reply:
Right, fixed